ethereumj-core/src/main/java/org/ethereum/sharding/util/Bitfield.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.sharding.util;
+
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.List;
+
+/**
+ * Bitfield is bit array where every bit represents status of
+ * attester with corresponding index
+ */
+public class Bitfield {
+
+    private final BitSet payload;
+    private final int size; // in Bits
+
+    private Bitfield(int size) {
+        this.size = calcLength(size) * Byte.SIZE;
+        this.payload = new BitSet(size);
+    }
+
+    public Bitfield(byte[] data) {
+        this.size = data.length * Byte.SIZE;
+        this.payload = BitSet.valueOf(data);
+    }
+
+    /**
+     * Calculates attesters bitfield length
+     * @param num  Number of attesters
+     * @return  Bitfield length in bytes
+     */
+    private static int calcLength(int num) {
+        return (num + 7) / Byte.SIZE;
```

Comment:
Consider an overflow check here. @Nashatyrev proposed a good and simple solution for such cases https://github.com/ethereum/ethereumj/blob/a207aeef0acbeb74e8c7d2bc5d248e4079673eea/ethereumj-core/src/main/java/org/ethereum/vm/VMUtils.java#L160-L162